### ISSUE TITLE
False positive non-related statements in try-block

### ISSUE THREAD

**Issue by Simbiat** [kalessil/phpinspectionsea#1945]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP 8.3                        |

Current behaviour
--
Have a code like this (simplified of [this file](https://github.com/Simbiat/database/blob/master/src/Database/Pool.php))
```php
try {
    $driver = 'driver';
    $connection = new \PDO('dsn', 'user', 'password', []);
    #Enforce some attributes. I've noticed that some of them do not apply when used during initial creation. The most frequent culprit is prepare emulation
    if ($driver === 'mysql') {
        if (!$connection->setAttribute(\PDO::MYSQL_ATTR_MULTI_STATEMENTS, false)) {
            throw new \PDOException('Failed to set `MYSQL_ATTR_MULTI_STATEMENTS` to `false`.');
        }
        if (!$connection->setAttribute(\PDO::MYSQL_ATTR_IGNORE_SPACE, true)) {
            throw new \PDOException('Failed to set `MYSQL_ATTR_IGNORE_SPACE` to `true`.');
        }
        if (!$connection->setAttribute(\PDO::MYSQL_ATTR_DIRECT_QUERY, false)) {
            throw new \PDOException('Failed to set `MYSQL_ATTR_DIRECT_QUERY` to `false`.');
        }
        if (!$connection->setAttribute(\PDO::MYSQL_ATTR_USE_BUFFERED_QUERY, true)) {
            throw new \PDOException('Failed to set `MYSQL_ATTR_USE_BUFFERED_QUERY` to `true`.');
        }
    } elseif ($driver === 'sqlsrv') {
        if (!$connection->setAttribute(\PDO::SQLSRV_ATTR_DIRECT_QUERY, false)) {
            throw new \PDOException('Failed to set `SQLSRV_ATTR_DIRECT_QUERY` to `false`.');
        }
    }
    if (!$connection->setAttribute(\PDO::ATTR_EMULATE_PREPARES, true)) {
        throw new \PDOException('Failed to set `ATTR_EMULATE_PREPARES` to `true`.');
    }
    if (!$connection->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION)) {
        throw new \PDOException('Failed to set `ATTR_ERRMODE` to exception mode.');
    }
} catch (\Throwable $exception) {
    #Do something
}
```
Get `[EA] Consider moving non-related statements (5 in total) outside the try-block or refactoring the try-body into a function/method.` warning on `try`.
Remove 
```php
    if (!$connection->setAttribute(\PDO::ATTR_EMULATE_PREPARES, true)) {
        throw new \PDOException('Failed to set `ATTR_EMULATE_PREPARES` to `true`.');
    }
    if (!$connection->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION)) {
        throw new \PDOException('Failed to set `ATTR_ERRMODE` to exception mode.');
    }
```
Lose the warning. Same behavior is observed with this code
```php
try {
    $driver = 'driver';
    $connection = new \PDO('dsn', 'user', 'password', []);
    #Enforce some attributes. I've noticed that some of them do not apply when used during initial creation. The most frequent culprit is prepare emulation
    if ($driver === 'mysql') {
        $connection->setAttribute(\PDO::MYSQL_ATTR_MULTI_STATEMENTS, false);
        $connection->setAttribute(\PDO::MYSQL_ATTR_IGNORE_SPACE, true);
        $connection->setAttribute(\PDO::MYSQL_ATTR_DIRECT_QUERY, false);
        $connection->setAttribute(\PDO::MYSQL_ATTR_USE_BUFFERED_QUERY, true);
    } elseif ($driver === 'sqlsrv') {
        $connection->setAttribute(\PDO::SQLSRV_ATTR_DIRECT_QUERY, false);
    }
    $connection->setAttribute(\PDO::ATTR_EMULATE_PREPARES, true);
    $connection->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
} catch (\Throwable $exception) {
    #Do something
}
```
But here it makes at least some sense, because `setAttribute` does not throw exceptions., but the number in the warning is still incorrect,  and it still reacts only to the `setAttribute` lines outside the if statements.

Expected behavior
--

No warning is shown, at least when there are exceptions thrown. If there are no exceptions in the code, number probably needs correction.

Environment details
---

```
PhpStorm 2024.1.1
Build #PS-241.15989.102, built on April 23, 2024
Licensed to simbiat.ru / Dmitry Kustov
Subscription is active until March 31, 2025.
For non-commercial open source development only.
Runtime version: 17.0.10+1-b1207.14 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
GC: G1 Young Generation, G1 Old Generation
Memory: 5120M
Cores: 16
Registry:
  debugger.new.tool.window.layout=true
  run.processes.with.pty=TRUE
  ide.experimental.ui=true
  terminal.new.ui=true
Non-Bundled Plugins:
  com.intellij.ml.llm (241.15989.156)
  com.kalessil.phpStorm.phpInspectionsEA (5.0.0.0)
  org.intellij.qodana (241.15989.121)
```

**Comment by ea-inspections-team** [comment]
Thank you for reporting @Simbiat!

The message wording could be better, but it suggesting to do a refactoring and extract parts of the try-body into a dedicated method or methods. This can be a good simplification, but depending on your project practices you might have to disable the inspection.

We will update the message for now. If we see more similar reports for this inspection, we'll revisit how it works.

**Comment by Simbiat** [comment]
I mean, it specifically suggests to remove non-related lines. When it suggests this and it's about some line like `$var = false;` or something that will not throw an exception - that's understandable. In my first example the issue is that there are actual throws, though, so it should not consider them as non-related. Unless it counts lines with `if` statements, but then it's unclear why it seems to ignore them (at least to an extent) in 2nd example.
If the message was saying "There are too many lines in try-catch" - then maybe it would make sense, but it's about "non-related" ones. If not redesign or rephrasing, this would require some clarification on what is considered as "non-related".

**Comment by ea-inspections-team** [comment]
Sorry for a possible confusion @Simbiat. `The message wording could be better` is referencing to inspection warning, which will be updated in the upcoming release.

**Comment by Simbiat** [comment]
Updated to what exactly? To `there are too many lines in try-catch`?

**Comment by ea-inspections-team** [comment]
To this one: `It is possible that some of the statements contained in the try block can be extracted into their own methods or functions (we recommend that you do not include more than three statements per try block).`

**Comment by Simbiat** [comment]
can we have a setting for this to change limit of 3 to any number? 3 feels a bit arbitrary, especially since at least in some cases `if` statement seems to be counted (and not the contents of the statement)

**Comment by ea-inspections-team** [comment]
Yes, we can add a setting in one of the following releases.
Could you please create a separate ticket for this feature request?

**Comment by Simbiat** [comment]
Done #1956

**Comment by ea-inspections-team** [comment]
Thank you @Simbiat.

We are pushing a commit which updates the message.
